Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -1795,15 +1829,18 @@ TextureCacheBase::TCacheEntry* TextureCacheBase::GetXFBFromCache(u32 address, u3
 
     // The only thing which has to match exactly is the stride. We can use a partial rectangle if
     // the VI width/height differs from that of the XFB copy.
-    if (entry->is_xfb_copy && entry->memory_stride == stride && entry->native_width >= width &&
-        entry->native_height >= height && !entry->may_have_overlapping_textures)
+    // One way this can happen is if there were oversized XFB copies that got reduced on
+    // creation.  Example: Teenage Mutant Ninja Turtles (Wii)
+    if (entry->is_xfb_copy && entry->memory_stride == stride && entry->native_width != width &&
+        entry->native_height != height && !entry->may_have_overlapping_textures)
     {
       // But if the dimensions do differ, we must compute the hash on the sub-rectangle.
       u64 check_hash = hash;
       if (entry->native_width != width || entry->native_height != height)
       {
-        check_hash = Common::GetHash64(Memory::GetPointer(entry->addr),
-                                       entry->memory_stride * entry->native_height, 0);
+        check_hash = CalculateTextureHash(
```

Comment:
You should be able to simplify this to just `check_hash = entry->CalculateHash();`

If not, we have another bug.

Reply:
We are using `entry->memory_stride * entry->native_height` here though.  If `native_height` does not equal `NumBlocksY` then this will be different than:  https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/VideoCommon/TextureCacheBase.cpp#L2905 .

Reply:
Ah perfect.  That does simplify this a lot.  I'll make that change.  Thanks phire!